sql-parser/src/main/java/io/crate/sql/tree/Privilege.java

```diff
@@ -0,0 +1,73 @@
+package io.crate.sql.tree;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class Privilege extends Node {
```

Comment:
Would it be possible to make Privilege an enum directly and not extend Node at all? Then the Grant/Revoke classes could be changed to take a Set<Privilege> (using an EnumSet as implementation). The `getALL()` case could also be made safer by using `EnumSet.allOf()` - this makes sure it always contains all values. As a bonus it would also require less memory (though I think that hardly matters in this case)

This would probably also lead to a slight change in the constructor of the Grant/Revoke classes - it would remove the boolean `allPrivileges` arg. Instead the Visitor would have to make sure to construct the Privilege enumSet correctly .

I think that's a good thing because from reading `GrantPrivilege(users, true, ...)` or `GrantPrivilege(users, false, ..)` it's not clear at all what the boolean is indicating